broker.go

```diff
@@ -1209,6 +1211,19 @@ func (c *offsetCoordinator) Offset(topic string, partition int32) (offset int64,
 	return 0, "", resErr
 }
 
+// Close releases the underlying connection back to the broker pool. Calling
+// clients should assume the coordinator cannot be re-used once it has been
+// closed.
+func (c *offsetCoordinator) Close() {
+	c.mu.Lock()
+	defer c.mu.Unlock()
+
+	if c.conn != nil && !c.conn.IsClosed() {
+		defer c.broker.conns.Idle(c.conn)
```

Comment:
You probably want to use `go` instead of `defer` here. The goal is not to block your return on `Idle` which can take some hundreds of milliseconds (whatever the re-enqueue wait is).